pymc4/distributions/continuous.py

```diff
@@ -1165,6 +1169,103 @@ def upper_limit(self):
         return self.conditions["high"]
 
 
+class Flat(ContinuousDistribution):
+    r"""A uniform distribution with support :math:`(-\inf, \inf)`.
+    Used as a uninformative log-likelihood that returns
+    zeros regardless of the passed values.
+    """
+
+    def __init__(self, name, **kwargs):
+        super().__init__(name, **kwargs)
+
+    @staticmethod
+    def _init_distribution(conditions):
+        return tfd.Uniform(low=-np.inf, high=np.inf)
+
+    def log_prob(self, value):
+        # convert the value to tensor
```

Comment:
I think that you're doing too many if statements. I recommend that you start out with `expected = tf.zeros(self._distribution.batch_shape + self._distribution.event_shape)`. This would be the expected shape of a call to a `tfd.Distribution.sample()`.

You should first check if `value` has a matching event shape part to `expected`, if yes then broadcast `expected` and `value` and create a zeros tensor with the resulting shape. Finally, you'll have to reduce sum over all event shape axes.